### DISCUSSION TITLE
Check Outputted List is Empty?

### DISCUSSION THREAD

**Discussion by merlinstardust** [community/community#27085] [Actions]
<p>Anyone know how I can check if an outputted list is empty in GitHub Actions?</p>
<p>I’m using <a href="https://github.com/dorny/paths-filter" rel="noopener nofollow ugc">dorny/paths-filter</a>, specifically the <code>use change detection to configure matrix job</code> example under <a href="https://github.com/dorny/paths-filter#conditional-execution" rel="noopener nofollow ugc">conditional execution</a> .</p>
<p>I was thinking of something like <code>if: join(steps.filter.outputs.packages)</code> and compare the joined result to an empty list, but I’m not sure if that would work or if there is a better way with some kind of hidden <code>.length</code> value</p>

**Comment by Simran-B** [comment, marked as answer]
<p>GitHub Actions expressions are fairly limited and I haven’t seen a length function to this day.</p>
<p>However, in this case you can simply compare the output, which is a JSON string, to the JSON string for an empty array:</p>
<pre><code class="lang-yaml">...
  build:
    needs: changes
    if: needs.changes.outputs.packages != '[]'
...
</code></pre>

**Comment by dmitry-stepanenko** [comment]
Might work slightly better to check the availability of the first element in the parsed array instead of a string
```
...
  build:
    needs: changes
    if: fromJson(needs.changes.outputs.packages)[0] != null
...
```

**Reply by Simran-B** [reply]
`[null]` is technically a non-empty array but wouldn't satisfy the condition. You probably get away with it most of the time, however.

**Reply by dmitry-stepanenko** [reply]
Yea, my use cases are not that generic, works great for an array of integers for example
